### PR TITLE
Implement callback for create_order

### LINKED ISSUE #88: Implement callback for create_order
Implement a function called create_order_callback in the polkadex gateway that accepts the incremental nonce provided by enclave to Openfinex during the initial API call to Openfinex and OrderUUID. Use that nonce to get and remove the order from the order cache and store it in the Orderbook Mirror using the provided OrderUUID

### PR DESCRIPTION
**Pull request by jaxter03**
Will close #88

### CODE REVIEW

File: enclave/src/polkadex_gateway.rs [hunk 1]
```diff
@@ -332,64 +336,85 @@ pub fn authenticate_user(
...
-//         return Err(sgx_status_t::SGX_ERROR_UNEXPECTED);
-//     } else {
-//         Ok(unsafe { &*ptr })
-//     }
-// }
+pub fn initialize_polkadex_gateway() {
+    // let nonce: u128 = 0;
+    // let create_nonce_storage_ptr = Arc::new(SgxMutex::<u128>::new(nonce));
+    // let create_nonce_ptr = Arc::into_raw(create_nonce_storage_ptr);
+    // CREATE_ORDER_NONCE.store(create_nonce_ptr as *mut (), Ordering::SeqCst);
```
Comment (Gauthamastro) [inline enclave/src/polkadex_gateway.rs]: @ksr30 Shouldn't these be uncommented?

**Comment by jaxter03** [comment]
@haerdib I need your help with RPC stuff ... I mentioned your name in the comments

**Review by Gauthamastro — APPROVED** [review]

File: enclave/src/tests.rs [hunk 2]
```diff
@@ -118,7 +119,7 @@ pub extern "C" fn test_main_entrance() -> size_t {
         // RPC API tests
         rpc_call_encoder::tests::test_encoding_none_params_returns_ok,
         rpc_get_balance::tests::test_given_valid_top_return_balances,
-        rpc_place_order::tests::test_given_valid_call_return_order_uuid,
+        //rpc_place_order::tests::test_given_valid_call_return_order_uuid, TODO: @Bigna this test case is failing ... I need your help with this
```
Comment (haerdib) [inline enclave/src/tests.rs:122]: we have fixed this in PR #95

File: enclave/src/rpc/rpc_place_order.rs [hunk 3]
```diff
@@ -65,15 +65,16 @@ impl RpcPlaceOrder {
...
-
-        let order_uuid =
-            match self
-                .rpc_gateway
-                .place_order(main_account, proxy_account.clone(), order)
-            {
-                Ok(uuid) => Ok(uuid),
-                Err(e) => Err(String::from(e.to_string())),
-            }?;
+        // FIXME @bigna I need your help to fix this file
```
Comment (haerdib) [inline enclave/src/rpc/rpc_place_order.rs:68]: Leave it like this. We have addressed this issue in PR #95

File: enclave/src/rpc/polkadex_rpc_gateway.rs [hunk 4]
```diff
@@ -109,7 +109,8 @@ impl RpcGateway for PolkadexRpcGateway {
         main_account: AccountId,
         proxy_acc: Option<AccountId>,
         order: Order,
-    ) -> Result<OrderUUID, GatewayError> {
+    ) -> Result<(), GatewayError> {
+        // TODO @gj @bigna please confirm this
```
Comment (haerdib) [inline enclave/src/rpc/polkadex_rpc_gateway.rs:113]: Confirmed.

**Review by haerdib — APPROVED** [review]
LGTM

**Pull request merged**